Lib/fontbakery/specifications/googlefonts.py

```diff
@@ -1600,6 +1605,322 @@ def check_font_version_fields(fb, ttFont):
+# com.google.fonts/test/142 - "Checking Google Greek Pro glyph coverage"
+# com.google.fonts/test/143 - "Checking Google Latin Core glyph coverage"
+# com.google.fonts/test/144 - "Checking Google Latin Expert glyph coverage"
+# com.google.fonts/test/145 - "Checking Google Latin Plus glyph coverage"
+# com.google.fonts/test/146 - "Checking Google Latin Plus (Optional Glyphs) glyph coverage"
+# com.google.fonts/test/147 - "Checking Google Latin Pro glyph coverage"
+# com.google.fonts/test/148 - "Checking Google Latin Pro (Optional Glyphs) glyph coverage"
+
+
+@register_test
+@old_style_test(
+    id='com.google.fonts/test/052'
+)
+def check_font_contains_all_required_tables(fb, ttFont):
+  """Font contains all required tables?"""
+  # See https://github.com/googlefonts/fontbakery/issues/617
+  tables = set(ttFont.reader.tables.keys())
+  glyphs = set(['glyf'] if 'glyf' in ttFont.keys() else ['CFF '])
+  if (REQUIRED_TABLES | glyphs) - tables:
+    missing_tables = [str(t) for t in (REQUIRED_TABLES | glyphs - tables)]
+    desc = (("Font is missing required "
+             "tables: [{}]").format(', '.join(missing_tables)))
+    if OPTIONAL_TABLES & tables:
+      optional_tables = [str(t) for t in (OPTIONAL_TABLES & tables)]
+      desc += (" but includes "
+               "optional tables [{}]").format(', '.join(optional_tables))
+    fb.error(desc)
+  else:
+    fb.ok("Font contains all required tables.")
+
+
+@register_test
+@old_style_test(
+    id='com.google.fonts/test/053'
+)
+def check_for_unwanted_tables(fb, ttFont):
+  """Are there unwanted tables?"""
+  unwanted_tables_found = []
+  for table in ttFont.keys():
+    if table in UNWANTED_TABLES:
+      unwanted_tables_found.append(table)
+
+  if len(unwanted_tables_found) > 0:
+    fb.error(("Unwanted tables were found"
+              " in the font and should be removed:"
+              " {}").format(', '.join(unwanted_tables_found)))
+  else:
+    fb.ok("There are no unwanted tables.")
+
+@register_condition
+@condition
+def ttfautohint_stats(font):
+  try:
+    import subprocess
+    import tempfile
+    hinted_size = os.stat(font).st_size
+
+    dehinted = tempfile.NamedTemporaryFile(suffix=".ttf", delete=False)
+    subprocess.call(["ttfautohint",
+                     "--dehint",
+                     font,
+                     dehinted.name])
+    dehinted_size = os.stat(dehinted.name).st_size
+    os.unlink(dehinted.name)
+  except OSError:
+    return {"missing": True}
+
+  return {
+    "dehinted_size": dehinted_size,
+    "hinted_size": hinted_size
+  }
+
+@register_test
+@old_style_test(
+    id='com.google.fonts/test/054'
+  , conditions=['ttfautohint_stats']
+)
+def check_hinting_filesize_impact(fb, font, ttfautohint_stats):
+  """Show hinting filesize impact.
+
+     Current implementation simply logs useful info
+     but there's no fail scenario for this checker."""
+
+  if "missing" in ttfautohint_stats.keys():
+    fb.warning("\n\n\nttfautohint is not available!"
```

Comment:
OH, so that error is about a missing `ttfautohint`, I don't like this implementation (will add a comment above). We could make an explicit test for it though, `has_ttfautohint` and probably a condition that calls `$ command -v ttfautohint` and checks the exit code.